backend/src/nodes/impl/rest.py

```diff
@@ -0,0 +1,95 @@
+import aiohttp
+import asyncio
+import base64
+import cv2
+import io
+import numpy as np
+import os
+from PIL import Image
+from enum import Enum
+from typing import Dict, Union
+
+from .image_utils import normalize
+from ..utils.utils import get_h_w_c
+
+STABLE_DIFFUSION_HOST = os.environ.get("STABLE_DIFFUSION_HOST", "127.0.0.1")
+STABLE_DIFFUSION_PORT = os.environ.get("STABLE_DIFFUSION_PORT", "7860")
+
+STABLE_DIFFUSION_TEXT2IMG_URL = (
+    f"http://{STABLE_DIFFUSION_HOST}:{STABLE_DIFFUSION_PORT}/sdapi/v1/txt2img"
+)
+STABLE_DIFFUSION_IMG2IMG_URL = (
+    f"http://{STABLE_DIFFUSION_HOST}:{STABLE_DIFFUSION_PORT}/sdapi/v1/img2img"
+)
+STABLE_DIFFUSION_INTERROGATE_URL = (
+    f"http://{STABLE_DIFFUSION_HOST}:{STABLE_DIFFUSION_PORT}/sdapi/v1/interrogate"
+)
+STABLE_DIFFUSION_OPTIONS_URL = (
+    f"http://{STABLE_DIFFUSION_HOST}:{STABLE_DIFFUSION_PORT}/sdapi/v1/options"
+)
+
+
+async def get_async(url) -> Dict:
+    async with aiohttp.ClientSession() as session:
+        async with session.get(url) as response:
+            return await response.json()
+
+
+async def post_async(url, json_data: Dict) -> Dict:
+    async with aiohttp.ClientSession() as session:
+        async with session.post(url, json=json_data) as response:
+            return await response.json()
```

Comment:
Do these methods have to be async? The only reason you had to add `AsyncNodeBase` and everything around it is because of these methods.

We run nodes in a separate thread pool, so it should be fine to use a sync blocking API, I believe. I'm not a python expert, so I could be wrong here.

Reply:
I'll investigate...  It seemed necessary at the time, but maybe I didn't have the full context.

Reply:
Oh yeah.  Absolutely unnecessary.  Works fine with `requests` instead of `aiohttp`.